CustomEnv Environment initialized
CustomEnv Environment initialized
/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/spaces/box.py:73: UserWarning: [33mWARN: Box bound precision lowered by casting to float32
  logger.warn(
I0526 11:54:40.517884 140371929474624 xla_bridge.py:455] Unable to initialize backend 'rocm': NOT_FOUND: Could not find registered platform with name: "rocm". Available platform names are: CUDA Interpreter Host
I0526 11:54:40.518450 140371929474624 xla_bridge.py:455] Unable to initialize backend 'tpu': module 'jaxlib.xla_extension' has no attribute 'get_tpu_client'
I0526 11:54:40.518763 140371929474624 xla_bridge.py:455] Unable to initialize backend 'plugin': xla_extension has no attributes named get_plugin_device_client. Compile TensorFlow with //tensorflow/compiler/xla/python:enable_plugin_device set to true (defaults to false) to enable this.
CustomEnv Environment reset
0it [00:00, ?it/s]
Traceback (most recent call last):
  File "/home/howard/RLproject/train_finetuning.py", line 230, in <module>
    app.run(main)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/absl/app.py", line 308, in run
    _run_main(main, args)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
  File "/home/howard/RLproject/train_finetuning.py", line 150, in main
    observation, done = env.reset(), False
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/wrappers/record_episode_statistics.py", line 20, in reset
    observations = super(RecordEpisodeStatistics, self).reset(**kwargs)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/core.py", line 292, in reset
    return self.env.reset(**kwargs)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/core.py", line 346, in reset
    return self.env.reset(**kwargs)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/core.py", line 320, in reset
    return self.observation(observation)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/wrappers/flatten_observation.py", line 13, in observation
    return spaces.flatten(self.env.observation_space, observation)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/functools.py", line 888, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/spaces/utils.py", line 95, in flatten_dict
    return np.concatenate([flatten(s, x[key]) for key, s in space.spaces.items()])
  File "/home/howard/anaconda3/envs/rlpd/lib/python3.9/site-packages/gym/spaces/utils.py", line 95, in <listcomp>
    return np.concatenate([flatten(s, x[key]) for key, s in space.spaces.items()])
TypeError: 'NoneType' object is not subscriptable